pkg/segment/tracing/utils/lineartimefinding_test.go

```diff
@@ -54,8 +55,12 @@ func TestFindPercentileData(t *testing.T) {
 		},
 	}
 
+	precision := 8
+	scale := math.Pow10(precision)
+
 	for _, tc := range testCases {
 		result := FindPercentileData(tc.arr, tc.percentile)
+		result = math.Round(result*scale) / scale
 		if result != tc.expected {
 			t.Errorf("Expected %d percentile to be %f, but got %f", tc.percentile, tc.expected, result)
```

Comment:
Just curious: was this test flaky for you?

Reply:
Yes it was, so the `result` is `29.040000000000003` but the `tc.expected` is `29.04`. So I had to set the precision.